aten/src/ATen/native/cuda/TensorTopK.cu

```diff
@@ -429,20 +652,31 @@ void launch(
   TORCH_CHECK(blocks_per_slice <= std::numeric_limits<uint32_t>::max(), "blocks_per_slice larger than uint32 maximum is not supported");
   auto semaphores_buffer = allocator.allocate(numInputSlices * sizeof(uint32_t));
   uint32_t* semaphores = reinterpret_cast<uint32_t*>(semaphores_buffer.get());
-  AT_CUDA_CHECK(cudaMemsetAsync(semaphores, 0, numInputSlices * sizeof(uint32_t), c10::cuda::getCurrentCUDAStream()));
+  AT_CUDA_CHECK(cudaMemsetAsync(semaphores, 0, numInputSlices * sizeof(uint32_t), stream));
 
-  auto ks_to_find_buffer = allocator.allocate(numInputSlices * sizeof(IndexType));
-  IndexType* ks_to_find = reinterpret_cast<IndexType*>(ks_to_find_buffer.get());
-  IndexType k_to_find = largest ? inputSliceSize - outputSliceSize + 1: outputSliceSize;
-  fill<IndexType><<<std::min((numInputSlices + 511) / 512, (IndexType)65535), 512, 0, c10::cuda::getCurrentCUDAStream()>>>(
+  auto ks_to_find_buffer = allocator.allocate(numInputSlices * sizeof(uint32_t));
+  uint32_t* ks_to_find = reinterpret_cast<uint32_t*>(ks_to_find_buffer.get());
+  uint32_t k_to_find = largest ? inputSliceSize - outputSliceSize + 1: outputSliceSize;
+  fill<uint32_t><<<std::min((numInputSlices + 511) / 512, (uint32_t)65535), 512, 0, stream>>>(
```

Comment:
nit - `min` is not needed here, blockDim.x can be 2**31?

Reply:
I think the largest value is `2**31-1`. For this line, probably the best is to use
```C++
std::min(((int64_t)numInputSlices + 511) / 512, (int64_t)some_large_number)
```
to handle overflow. I updated this PR with it.